frontend/test/stores/documentUnitStore.spec.ts

```diff
@@ -17,6 +17,50 @@ describe("useDocumentUnitStore", () => {
     vi.resetAllMocks()
   })
 
+  describe("remove TextCheck tags from given text", () => {
+    it("removes text-check tags from text", async () => {
```

Comment:
Okay. This is not the right test! You mock everything and test the mocked response. Makes no sense.
I think it would be better if you check the patch which is sent to the backend.

Something like `expect(documentUnitServiceMock).toHaveBeenCalledWith(<patch>)`!